src/View/Helper/PhoneNumberFormat.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\I18n\PhoneNumber\View\Helper;
+
+use Laminas\I18n\PhoneNumber\Exception\ExceptionInterface;
+use Laminas\I18n\PhoneNumber\Exception\InvalidArgumentException;
+use Laminas\I18n\PhoneNumber\PhoneNumberValue;
+
+use function preg_match;
+use function strtoupper;
+
+final class PhoneNumberFormat
+{
+    /** @var non-empty-string|null */
+    private ?string $country;
+
+    /**
+     * @param non-empty-string|null $defaultCountryCode
+     * @throws InvalidArgumentException If the country code is not a valid ISO 3166 string.
+     */
+    public function __construct(?string $defaultCountryCode = null)
+    {
+        $this->country = self::assertValidCountryCode($defaultCountryCode);
+    }
```

Comment:
With this new handling of the country code we have to deprecate the current setting and handling of a locale in all existing view helpers of laminas-view.

Example: https://docs.laminas.dev/laminas-i18n/view-helpers/number-format/#using-locale

Otherwise we create an inconsistency.

Reply:
This has got me thinking that a CountryCode value object would be more useful instead of dragging around and validating the string value all the time… Something like:

```php
final class CountryCode
{
  private function __construct(public readonly string $code)
  {
  }

  public static function fromLocaleString(string $locale): self;

  public static function fromString(string $code): self;

  public static function fromLocale(Locale $locale): self;

  public static function fromMixed(mixed $value): self;
}
```

This would be more flexible and provide better compat with i18n right? i.e. the view helper constructor would then become…

```php
public function __construct(?mixed $localeOrCountryCode = null)
{
  $localeOrCountryCode = $localeOrCountryCode ?: Locale::getDefault();
  $this->countryCode = CountryCode::fromMixed($localeOrCountryCode);
}
```

WDYT?

Reply:
> …a CountryCode value object would be more useful instead of dragging around and validating the string value all the time…

The value object is a good idea and the correct way to go. 👍